### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by Steve-Mcl**
## Description

Fix up tests, ensure they are actually ran, re-jig resources for re-use and common practices. Add 2nd test to ensure reusability. 

* [rename the shared ui prop, expose it for testing](https://github.com/flowforge/flowforge-nr-dashboard/commit/69c4ef19ca1d5d488a0f4aa69ac9343e0c534823)
* [correct handling of RED.settings.httpNodeRoot](https://github.com/flowforge/flowforge-nr-dashboard/commit/82f63be268f15942c84403022a4ad94815acf9f6)
* [dont crash if theme not set](https://github.com/flowforge/flowforge-nr-dashboard/commit/427a46ac537c56188f3f4f8165cfc6c810e3eec2)
* [dont crash if group not set](https://github.com/flowforge/flowforge-nr-dashboard/commit/19aafdeac591dfb649f8170e340d0152aa67081b)
* [add missing fixtures](https://github.com/flowforge/flowforge-nr-dashboard/commit/08e7b84fe48ce35b930625d60755da70ebcd2028)
* [update page fixture](https://github.com/flowforge/flowforge-nr-dashboard/commit/1003b0b37fd4fe36c16e902af835401cf4d11e16)
* [fix up button tests and re-jig reusable code](https://github.com/flowforge/flowforge-nr-dashboard/commit/0029ec34e9852407c9b5ecab18d1f3efda695a5b)
* [add another test to check re-jig is suitable](https://github.com/flowforge/flowforge-nr-dashboard/commit/0aa7651f8ef8ef9f74f6cee747c8cd50c6283e5a)

## Related Issue(s)

Cannot find  issue 👀 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by joepavitt** [comment]
> [rename the shared ui prop, expose it for testing](https://github.com/flowforge/flowforge-nr-dashboard/commit/69c4ef19ca1d5d488a0f4aa69ac9343e0c534823)

Can you explain in some more detail why we've now got `uiShared` please, instead of just `ui`?

File: .github/workflows/test.yml
Comment (ppawlowski) [inline .github/workflows/test.yml:1]: Testing is already covered in [the publish pipeline](https://github.com/flowforge/flowforge-nr-dashboard/pull/139). This workflow is not needed.

**Comment by Steve-Mcl** [comment]
> > [rename the shared ui prop, expose it for testing](https://github.com/flowforge/flowforge-nr-dashboard/commit/69c4ef19ca1d5d488a0f4aa69ac9343e0c534823)
> 
> Can you explain in some more detail why we've now got `uiShared` please, instead of just `ui`? Not sure I understand the `shared` nature of the variable you're referring to?

I will better describe this in our catch up but TLDR; In order to mock the connection, I needed access to the shared `ui` object for testing and so added it to each `node` - thus enabling (in tests) the ability to add a mocked connection to `node.ui` HOWEVER, there was already a `node.ui` (containing all the dashboards and widgets etc) so I renamed it for distinction (e.g. it is effectively shared across all instances as it is declared outside of a node and it holds connections for all dashboards etc).

It seemed necessary when I was fixing up the tests but in actuality, I ended up not using it. BUT I left it because there should be tests that check for ui-updates being transmitted to all connected - these emit on the connections inside the `uiShared` object.

File: .github/workflows/test.yml
Comment (joepavitt) [reply .github/workflows/test.yml:1]: Missed that - thanks

Comment (ppawlowski) [reply .github/workflows/test.yml:1]: It is happening as you described by using this `if` statement: https://github.com/flowforge/flowforge-nr-dashboard/actions/runs/6009791454/workflow#L20

**Review by joepavitt — APPROVED** [review]

**Pull request merged**